test-main.js

```diff
@@ -0,0 +1,65 @@
+var allTestFiles = ['assert', 'underscore'];
+var TEST_REGEXP = /tests\/.*\.js$/i;
+
+// Automatically clenaup sinon spies and stubs.
+var realSinon = sinon;
+beforeEach(function() {
+    sinon = realSinon.sandbox.create();
+});
+afterEach(function() {
+    sinon.restore();
+    sinon = realSinon;
+});
+
+function withSettings(changes, test) {
```

Comment:
Good call on the synchronous thing. Async will cause some issues here. That's a definite benefit to Squire. I'll see if there's something simple to support async but I'm not crazy about having two different ways of calling this (like the `done` for `it` but just another place to remember it).